build_tools/cmake/build_and_test_byo_llvm.sh

```diff
@@ -24,6 +24,11 @@ echo "Setting up venv at $VENV_DIR"
 python3 -m venv "$VENV_DIR"
 source "$VENV_DIR/bin/activate"
 python -m pip install -r runtime/bindings/python/iree/runtime/build_requirements.txt
+# Note: In a bundled build, pybind is made available via FetchContent.
+# However, if bringing your own LLVM, MLIR is configured before IREE is and
+# therefore pybind wasn't fetched and properly set up yet. To work around this,
+# pybind is installed via pip.
+python -m pip install pybind11==2.13.6
```

Comment:
How about we follow https://mlir.llvm.org/docs/Bindings/Python/#building and run the suggested
```bash
python -m pip install -r mlir/python/requirements.txt
```
here instead?

The "bring-your-own-LLVM" path tested here is one demonstration of how users can build LLVM before IREE. Needing to install dependencies isn't so much "working around an improperly set up build" as it is "following the upstream instructions". Note that https://github.com/llvm/llvm-project/blob/main/mlir/python/requirements.txt sets `pybind11>=2.9.0, <=2.10.3`. If that isn't compatible with _our_ build then we could send a patch upstream to increase the maximum version... or keep this local pin and add a comment explaining why we don't use the upstream requirements.

We could also add a comment on prerequisites to https://github.com/iree-org/iree/blob/main/build_tools/llvm/byo_llvm.sh since that unconditionally enables the upstream Python bindings: https://github.com/iree-org/iree/blob/9731fed369ef59ee1e120f9f1315a7b9981e515c/build_tools/llvm/byo_llvm.sh#L116

Reply:
Installing by
```
python -m pip install -r mlir/python/requirements.txt
```
would get us
```
pybind11>=2.9.0, <=2.10.3
```
Probably not the best idea to mix versions?

Reply:
How about we follow the upstream instructions _then also_ manually upgrade to the version we want, with a comment / TODO to relax the requirement upstream. Then ideally also send a PR upstream to resolve that.

Reply:
I revised the notes, but anyway we install the same version we use explicitly via pip.